source/expressions.tex

```diff
@@ -5042,10 +5042,11 @@
 \pnum
 \indextext{\idxcode{noexcept}}%
 \indextext{expression!\idxcode{noexcept}}%
-The \keyword{noexcept} operator determines whether the evaluation of its operand,
-which is an unevaluated operand\iref{term.unevaluated.operand}, can throw an
-exception\iref{except.throw}.
-
+The \keyword{noexcept} operator yields \keyword{true}
+unless the full-expression of the operand
+is potentially-throwing\iref{except.spec}.
+The operand of the \keyword{noexcept} operator
+is an unevaluated operand\iref{term.unevaluated.operand}.
 \begin{bnf}
```

Comment:
I think there is some room for improvement in this subclause, but talking about "the operand" in detail before we've even introduced the grammar seems unhelpful.

My suggestion:
- grammar first
- then talk about the operand
- then talk about the result
- then have the note about constant expression

Reply:
I've implemented most of what you've suggested, but I don't understand what you mean by "constant expression". Neither the old wording nor the new wording talks about constant expressions.

Reply:
"[Note 1 : A noexcept-expression is an integral constant expression (7.7). — end note]"

Reply:
Ah yes, I wasn't looking at the second paragraph. I've moved it up now.